encode.go

```diff
@@ -24,6 +24,35 @@ import (
 	perrors "github.com/pkg/errors"
 )
 
+var listTypeName = map[string]string{
+	"string": "[string",
```

Comment:
should be [java.lang.String?

Reply:
No, it is [string not [Ljava.lang.String

Reply:
@wongoo After I tried, I found that they all could.